models/templateBD.go

```diff
@@ -6,26 +6,47 @@ type TemplateBD struct {
 	Value map[string]interface{} `json:",omitempty"`
 }
 
-func NewTemplateBD(ops, path, name, displayName, layer2Unicast string, intersiteBumTrafficAllow, optimizeWanBandwidth, l2Stretch, l3MCast bool, vrfRef, dhcpLabel map[string]interface{}) *TemplateBD {
+func NewTemplateBD(ops, path, name, displayName, layer2Unicast, unkMcastAct, multiDstPktAct, v6unkMcastAct, vmac string, intersiteBumTrafficAllow, optimizeWanBandwidth, l2Stretch, l3MCast, arpFlood, unicastRouting bool, vrfRef, dhcpLabel map[string]interface{}) *TemplateBD {
 	var bdMap map[string]interface{}
 	bdMap = map[string]interface{}{
 		"name":                     name,
 		"displayName":              displayName,
 		"l2UnknownUnicast":         layer2Unicast,
+		"unkMcastAct":              unkMcastAct,
+		"multiDstPktAct":           multiDstPktAct,
+		"v6unkMcastAct":            v6unkMcastAct,
+		"vmac":                     vmac,
+		"arpFlood":                 arpFlood,
+		"unicastRouting":           unicastRouting,
 		"intersiteBumTrafficAllow": intersiteBumTrafficAllow,
 		"optimizeWanBandwidth":     optimizeWanBandwidth,
 		"l2Stretch":                l2Stretch,
 		"l3MCast":                  l3MCast,
 		"vrfRef":                   vrfRef,
 		"dhcpLabel":                dhcpLabel,
-		"subnets":                  []interface{}{},
 	}
 
 	if bdMap["l2UnknownUnicast"] == "" {
 		bdMap["l2UnknownUnicast"] = "flood"
 	}
 
-	if bdMap["dhcpLabel"] == nil {
+	if bdMap["unkMcastAct"] == "optimized_flooding" {
+		bdMap["unkMcastAct"] = "opt-flood"
+	} else {
+		bdMap["unkMcastAct"] = "flood"
+	}
+
+	if bdMap["multiDstPktAct"] == "flood_in_bd" || bdMap["multiDstPktAct"] == "" {
+		bdMap["multiDstPktAct"] = "bd-flood"
+	}
+
+	if bdMap["v6unkMcastAct"] == "optimized_flooding" {
+		bdMap["v6unkMcastAct"] = "opt-flood"
+	} else {
+		bdMap["v6unkMcastAct"] = "flood"
```

Comment:
By setting an implicit default this might not work on MSO that do not support those. Did you test on MSO that don't have those options? Also we could move the else into a else if and avoid that issue.

Reply:
Yes, tested on an older version of MSO and it works fine.